doc/reference/networking/can_api.rst

```diff
@@ -145,10 +154,10 @@ a mailbox. When a transmitting mailbox is assigned, sending cannot be canceled.
   struct zcan_frame frame = {
           .id_type = CAN_STANDARD_IDENTIFIER,
           .rtr = CAN_DATAFRAME,
-          .ext_id = 0x123,
+          .id = 0x123,
           .dlc = 8
   };
-  struct device *can_dev;
+  const struct device *can_dev;
   int ret;
 
   frame.data = {1,2,3,4,5,6,7,8};
```

Comment:
```suggestion
  frame.data[0] = 1;
  frame.data[1] = 2;
  frame.data[2] = 3;
  frame.data[3] = 4;
  frame.data[4] = 5;
  frame.data[5] = 6;
  frame.data[6] = 7;
  frame.data[7] = 8;
```

Not sure, if it is my issue, but with this line I get: 

```
error: expected expression before '{' token
  <line_number> |  frame.data = {1, 2, 3, 4, 5, 6, 7, 8};
```

My suggestion fixes it.

Reply:
Moved it to initializer.